src/pathOrderOptimizer.h

```diff
@@ -17,17 +37,15 @@ namespace cura {
 class PathOrderOptimizer
 {
 public:
-    EZSeamType type;
     Point startPoint; //!< A location near the prefered start location
-    Point z_seam_pos; //!< The position near where to create the z_seam (if \ref PathOrderOptimizer::type == 'back')
+    const ZSeamConfig* config;
     std::vector<ConstPolygonPointer> polygons; //!< the parts of the layer (in arbitrary order)
     std::vector<int> polyStart; //!< polygons[i][polyStart[i]] = point of polygon i which is to be the starting point in printing the polygon
     std::vector<int> polyOrder; //!< the optimized order as indices in #polygons
 
-    PathOrderOptimizer(Point startPoint, Point z_seam_pos = Point(0, 0), EZSeamType type = EZSeamType::SHORTEST)
-    : type(type)
-    , startPoint(startPoint)
-    , z_seam_pos(z_seam_pos)
+    PathOrderOptimizer(Point startPoint, const ZSeamConfig* config = nullptr)
```

Comment:
Following the comment above I'd like this function signature to be
```
PathOrderOptimizer(Point startPoint, const ZSeamConfig& config)
```
or
```
PathOrderOptimizer(Point startPoint, const ZSeamConfig& config = ZSeamConfig())
```
So you can just call it with
```
PathOrderOptimizer optimizer(startPoint, ZSeamConfig())
```
or
```
PathOrderOptimizer optimizer(startPoint)
```